server/src/controllers/document.ts

```diff
@@ -75,7 +75,7 @@ export const searchDocuments = async (req: Request, res: Response) => {
     }
     
     // Create embedding for the query
-    const queryEmbedding = await createEmbedding(query);
+    const queryEmbedding = await createEmbedding(query, 'retrieval_query');
```

Comment:
we'd better use enum for type safe purpose

```suggestion
    const queryEmbedding = await createEmbedding(query, TaskType.RETRIEVAL_QUERY);
```